spec.bs

```diff
@@ -1701,3 +1748,147 @@ specification is printed below:
+the permissions [=fenced frame config/required permissions to load|required=] for a <{fencedframe}>
+to navigate successfully. Specifically, a {{Document}} inside of a <{fencedframe}> can only load if
+it has opted into all of the [=fenced frame config/required permissions to load=].
+
+<h4 id=fenced-container-policies>Fenced container policies</h4>
+
+TODO: define <dfn>fenced container policy</dfn> here, which is current referenced by
+<{fencedframe/allow}>.
+
+<h4 id=permissions-policy-patches>Algorithm patches</h4>
+
+<div algorithm=create-permissions-policy>
+  Modify the definition of [=Create a Permissions Policy for a navigable=] to read:
+
+  Given null or an element (|container|), an [=origin=] (<var ignore>origin</var>), and an optional
+  [=list=] of [=policy-controlled features=] (|required features|), this algorithm returns a new
+  [=Document/permissions policy=].
+
+  Rewrite step 1 of the algorithm to read:
+
+  1. [=Assert=]: if not null, |container| is either a [=navigable container=] or a [=fenced
+     navigable container=].
+
+  Add a new step after step 1 that reads:
+
+  2. [=Assert=]: if |container| is not a [=fenced navigable container=], |required features| is not
+     given.
+
+  Rewrite step 4 to read:
+
+  4. If |container| is a [=fenced navigable container=], then for each |feature| supported:
+
+       1. If |feature| [=list/exists=] in |required features|, set |inherited policy|[|feature|] to
+          "`Enabled`".
+
+          Otherwise, set |inherited policy|[|feature|] to "`Disabled`".
+
+     Otherwise, for each |feature| supported,
+
+       1. Let |isInherited| be the result of running [=Define an inherited policy for feature in
+          container at origin=] on |feature|, |container| and <var ignore>origin</var>.
+
+       2. Set |inherited policy|[|feature|] to |isInherited|.
+</div>
+
+<div algorithm=allow-attribute-fenced-frame>
+  Rename the <a href=https://w3c.github.io/webappsec-permissions-policy/#iframe-allow-attribute>The
+  `allow` attribute of the `iframe` element</a> section to "The `allow` attribute of the `iframe`
+  and `fencedframe` element", and rewrite the section to read:
+
+  <{iframe}> and <{fencedframe}> elements have an respective `allow` attributes (<{iframe}>:
+  <{iframe/allow}>; <{fencedframe}>: <{fencedframe/allow}>), which contain an [=ASCII-serialized
+  policy directive=].
+
+  The allowlist for the features named in the attribute may be empty; in that case, the default
+  value for the allowlist is "`src`", which represents the origin of the URL in the iframe’s src
+  attribute, or the fencedframe's [=fenced frame config=].
+
+  When not empty, the <{iframe/allow}> attribute will result in adding an allowlist for each
+  recognized feature to the iframe element’s content navigable's container policy or the fencedframe
+  element's [=fenced navigable container/fenced navigable=]'s container policy, when it is
```

Comment:
All of these references to container policy should probably be updated to point to either the already-existing "container policy" or the new "fenced container policy" when appropriate. This all assumes that we should introduce the concept of a "fenced container policy" that's independent of "container policy", instead of just modifying the existing "container policy definition accordingly. I'll let you decide which is best to do.

Reply:
My vote is to stick with "container policy". The container policies themselves behave the exact same between fenced frames and iframes. The difference is how external algorithms handle them. So I think we'd just be making a duplicate of "container policy" with "fenced container policy".